protocol/messenger_response.go

```diff
@@ -9,70 +11,199 @@ import (
+	for _, c := range r.communities {
+		communities = append(communities, c)
+	}
+	return communities
 }
 
 func (m *MessengerResponse) IsEmpty() bool {
-	return len(m.Chats)+len(m.Messages)+len(m.Contacts)+len(m.Installations)+len(m.Invitations)+len(m.EmojiReactions)+len(m.Communities)+len(m.CommunityChanges)+len(m.Filters)+len(m.RemovedFilters)+len(m.RemovedChats)+len(m.Notifications)+len(m.MailserverTopics)+len(m.Mailservers)+len(m.MailserverRanges) == 0
+	return len(m.chats)+len(m.Messages)+len(m.Contacts)+len(m.Installations)+len(m.Invitations)+len(m.EmojiReactions)+len(m.communities)+len(m.CommunityChanges)+len(m.Filters)+len(m.RemovedFilters)+len(m.removedChats)+len(m.MailserverTopics)+len(m.Mailservers)+len(m.MailserverRanges)+len(m.Notifications)+len(m.RequestsToJoinCommunity) == 0
 }
 
 // Merge takes another response and appends the new Chats & new Messages and replaces
 // the existing Messages & Chats if they have the same ID
 func (m *MessengerResponse) Merge(response *MessengerResponse) error {
-	if len(response.Contacts)+len(response.Installations)+len(response.EmojiReactions)+len(response.Invitations) != 0 {
+	if len(response.Contacts)+len(response.Installations)+len(response.EmojiReactions)+len(response.Invitations)+len(response.RequestsToJoinCommunity)+len(response.Mailservers)+len(response.MailserverTopics)+len(response.MailserverRanges)+len(response.Notifications)+len(response.EmojiReactions)+len(response.CommunityChanges) != 0 {
 		return ErrNotImplemented
 	}
 
-	m.overrideChats(response.Chats)
+	m.AddChats(response.Chats())
+	m.AddRemovedChats(response.RemovedChats())
 	m.overrideMessages(response.Messages)
+	m.overrideFilters(response.Filters)
+	m.overrideRemovedFilters(response.Filters)
+	m.AddCommunities(response.Communities())
 
 	return nil
 }
 
-// overrideChats append new chats and override existing ones in response.Chats
-func (m *MessengerResponse) overrideChats(chats []*Chat) {
-	for _, overrideChat := range chats {
+// overrideMessages append new messages and override existing ones in response.Messages
+func (m *MessengerResponse) overrideMessages(messages []*common.Message) {
+	for _, overrideMessage := range messages {
 		var found = false
-		for idx, chat := range m.Chats {
-			if chat.ID == overrideChat.ID {
-				m.Chats[idx] = overrideChat
+		for idx, chat := range m.Messages {
+			if chat.ID == overrideMessage.ID {
+				m.Messages[idx] = overrideMessage
 				found = true
 			}
 		}
 		if !found {
-			m.Chats = append(m.Chats, overrideChat)
+			m.Messages = append(m.Messages, overrideMessage)
 		}
 	}
 }
 
-// overrideMessages append new messages and override existing ones in response.Messages
-func (m *MessengerResponse) overrideMessages(messages []*common.Message) {
-	for _, overrideMessage := range messages {
+// overrideFilters append new filters and override existing ones in response.Filters
+func (m *MessengerResponse) overrideFilters(filters []*transport.Filter) {
```

Comment:
overrideFilters and overrideRemovedFilters are identical except used used field. Can be implemented as single function with pointer to field passed